src/rpc/methods/eth/utils.rs

```diff
@@ -97,6 +100,130 @@ where
     }
 }
 
+/// Extract and decode Ethereum revert reason from receipt return data
+pub fn decode_revert_reason(return_data: RawBytes) -> (Vec<u8>, String) {
+    let (data, reason) = match decode_payload(&return_data, CBOR) {
+        Err(e) => {
+            log::warn!("failed to unmarshal cbor bytes from message receipt return error: {e}");
+            (
+                EthBytes::default(),
+                String::from("ERROR: revert reason is not cbor encoded bytes"),
+            )
+        }
+        Ok(data) => (data.clone(), parse_eth_revert(data.0.as_slice())),
+    };
+
+    (data.0, reason)
+}
+
+const ERROR_FUNCTION_SELECTOR: [u8; 4] = [0x08, 0xc3, 0x79, 0xa0]; // Error(string)
```

Comment:
Could you explain how to obtain those magic bytes?

Reply:
Good, maybe put in the comment i.e. `... 0xa0]; // keccak256("Error(string)")`

Reply:
Done